components/post_view/post_view.jsx

```diff
@@ -0,0 +1,393 @@
+         * Used for taking a backup before clearing from redux-store.
+         */
+        postIdsInCurrentChannel: PropTypes.array,
+
+        focusedPostId: PropTypes.string,
+
+        actions: PropTypes.shape({
+
+            /*
+             * Get post for permaview
+             */
+            getPostThread: PropTypes.func.isRequired,
+
+            /*
+             * Get unreads posts onload
+             */
+            loadUnreads: PropTypes.func.isRequired,
+
+            /*
+             * Get posts using BEFORE_ID and AFTER_ID
+             */
+            loadPosts: PropTypes.func.isRequired,
+
+            /*
+             * Used for changing posts status of channel
+             * For notifying redux store when we hit bottom or top of posts
+             */
+            changeChannelPostsStatus: PropTypes.func.isRequired,
+
+            /*
+             * Used for notifying redux store on channel sync complete
+             */
+            channelSyncCompleted: PropTypes.func.isRequired,
+
+            /*
+             * Used to sync when channel is out of sync because of socket disconnect
+             */
+            syncChannelPosts: PropTypes.func.isRequired,
+
+        }).isRequired,
+    }
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            isDoingInitialLoad: this.shouldLoadPosts(props),
+            newerPosts: {
+                loading: false,
+                allLoaded: (props.channelPostsStatus && props.channelPostsStatus.atEnd) || false,
+            },
+            olderPosts: {
+                loading: false,
+                allLoaded: PostView.oldestMessageLoadedInView(props),
+            },
+            lastViewedAt: props.lastViewedAt,
+            posts: this.props.posts,
+        };
+    }
+
+    componentDidMount() {
+        if (this.state.isDoingInitialLoad) {
+            this.postsOnLoad(this.props.channelId);
+        }
+        if (!this.props.channelSyncStatus && this.props.channelPostsStatus) {
+            this.syncChannelPosts(this.props);
+        }
+    }
+
+    static getDerivedStateFromProps(nextProps, prevState) {
+        if (prevState.posts && nextProps.posts && (prevState.posts.length !== nextProps.posts.length)) {
+            if (PostView.oldestMessageLoadedInView(nextProps) !== prevState.olderPosts.allLoaded) {
+                return {
+                    olderPosts: {
+                        loading: false,
+                        allLoaded: PostView.oldestMessageLoadedInView(nextProps),
+                    },
+                    posts: nextProps.posts,
+                };
+            }
+            return {
+                posts: nextProps.posts,
+            };
+        }
+        return null;
+    }
+
+    componentDidUpdate(prevProps) {
+        if (this.props.socketStatus && this.props.socketStatus.connected && !prevProps.socketStatus.connected) {
+            this.syncChannelPosts(this.props);
```

Comment:
Continuing [this conversation](https://github.com/mattermost/mattermost-webapp/pull/1464#discussion_r206212158), is this component unmounted and remounted when changing channels? If that's not the case, then the mount logic wouldn't handle syncing posts on channel change.

Reply:
Yes. The component gets mounted on channel switch. so this gets handled on componentDidmount.